packages/core/test/Escrow.ts

```diff
@@ -2,68 +2,37 @@ import { anyValue } from '@nomicfoundation/hardhat-chai-matchers/withArgs';
 import { expect } from 'chai';
 import { ethers } from 'hardhat';
 import { Signer } from 'ethers';
-import { Escrow, HMToken } from '../typechain-types';
-
-const MOCK_URL = 'http://google.com/fake';
-const MOCK_HASH = 'kGKmnj9BRf';
-const BULK_MAX_COUNT = 100;
-
-enum Status {
-  Launched = 0,
-  Pending = 1,
-  Partial = 2,
-  Paid = 3,
-  Complete = 4,
-  Cancelled = 5,
-}
-
-let owner: Signer,
-  launcher: Signer,
-  reputationOracle: Signer,
-  recordingOracle: Signer,
-  externalAddress: Signer,
-  restAccounts: Signer[],
-  trustedHandlers: string[];
-
-let token: HMToken, escrow: Escrow;
-
-async function deployEscrow() {
-  // Deploy Escrow Contract
-  const Escrow = await ethers.getContractFactory('Escrow');
-  escrow = await Escrow.deploy(
-    token.address,
-    await owner.getAddress(),
-    100,
-    trustedHandlers
-  );
-}
-
-async function setupEscrow() {
-  await escrow
-    .connect(owner)
-    .setup(
-      await reputationOracle.getAddress(),
-      await recordingOracle.getAddress(),
-      10,
-      10,
-      MOCK_URL,
-      MOCK_HASH
-    );
-}
-
-async function addTrustedHandlers() {
-  await escrow
-    .connect(owner)
-    .addTrustedHandlers([await reputationOracle.getAddress()]);
-}
-
-async function fundEscrow() {
-  const amount = 100;
-  await token.connect(owner).transfer(escrow.address, amount);
-}
+import { Escrow, HMToken, RewardPool, Staking } from '../typechain-types';
 
 describe('Escrow', function () {
-  this.beforeAll(async () => {
+  const MOCK_URL = 'http://google.com/fake';
+  const MOCK_HASH = 'kGKmnj9BRf';
+  const BULK_MAX_COUNT = 100;
+
+  enum Status {
+    Launched = 0,
+    Pending = 1,
+    Partial = 2,
+    Paid = 3,
+    Complete = 4,
+    Cancelled = 5,
+  }
+
+  let owner: Signer,
+    launcher: Signer,
+    reputationOracle: Signer,
+    recordingOracle: Signer,
+    externalAddress: Signer,
+    restAccounts: Signer[],
+    trustedHandlers: string[];
+
+  let token: HMToken, escrow: Escrow, staking: Staking, rewardPool: RewardPool;
+
+  const minimumStake = 2;
+  const lockPeriod = 2;
+  const rewardFee = 2;
+
+  beforeEach(async () => {
```

Comment:
it should be beforeAll, you don't need to deploy a new factory for each test

Reply:
Originally, there was deploy helper functions used for each `describe` within `before` function. Now, we use `beforeEach` instead of `beforeAll`, and I believe it's doing the same thing for testing. For both cases, we deploy all the contracts before each test case. I prefer using `beforeEach` while it makes more sense, and we don't need unnecessary helper functions to deploy. If this makes sense, we can resolve the comments above about the eliminated helper functions.

Reply:
using beforeAll we reduce the execution time of the tests, we have optimize this, since using a monorepo the number of tests grows a lot